### PR TITLE
test visibility: add Python manual instrumentation

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
### What does this PR do? What is the motivation?

Adds documentation for the manual test visibility instrumentation API being added in Python `ddtrace` version `2.13` .

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/tests/setup/python.md [hunk 1]
```diff
@@ -163,6 +163,213 @@ assert True
...
+
+### Using manual testing API
+
+<div class="alert alert-warning"><strong>Note</strong>: The Test Visibility Manual API is in <strong>beta</strong> and subject to change.</div>
+
+As of version `2.13.0`, the [Datadog Python tracer][1] provides the Test Visibility API (`ddtrace.ext.test_visibility`) as a way to submit test visibility results in an ad-hoc manner.
+
+#### API execution
+
+The API uses classes to provide namespaced methods that are intended to be used without the caller maintaining state.
```
Comment (vitor-de-araujo) [inline content/en/tests/setup/python.md:176]: I'm not sure what "without the caller maintaining state" means here, and I'm not sure if it means much for a user of the library.

**Review by vitor-de-araujo — APPROVED** [review]

**Review by juan-fernandez — APPROVED** [review]

**Comment by romainkomorndatadog** [comment]
@drichards-87 , is this still pending review?

**Comment by drichards-87** [comment]
@romainkomorndatadog Yes, it look like it is. The Jira card was assigned to @alai97. I'd recommend reaching out to him for an update on the status.

File: content/en/tests/setup/python.md [hunk 2]
```diff
@@ -163,6 +163,213 @@ assert True
...
+Call `ddtrace.ext.test_visibility.api.TestModule.start()`, passing the `TestModuleId` object as an argument, to start the module.
+
+After all the children items within the module have completed, call `ddtrace.ext.test_visibility.api.TestModule.finish()`, passing the `TestModuleId` object as an argument.
+
+
+##### Test suite
+
+A test suite represents a subset of tests within a project's modules (`.py` file, for example).
+
+Call `ddtrace.ext.test_visibility.api.TestSuiteId()`, providing the parent module's `TestModuleId`, and the suite's name, as arguments, to create a `TestSuiteId`.
```
Comment (urseberry) [inline content/en/tests/setup/python.md:226]:
```suggestion
Call `ddtrace.ext.test_visibility.api.TestSuiteId()`, providing the parent module's `TestModuleId` and the suite's name as arguments, to create a `TestSuiteId`.
```

File: content/en/tests/setup/python.md [hunk 3]
```diff
@@ -163,6 +163,213 @@ assert True
...
+
+Call `ddtrace.ext.test_visibility.api.TestSuite.discover()`, passing the `TestSuiteId` object as an argument, to discover the suite.
+
+Call `ddtrace.ext.test_visibility.api.TestSuite.start()`, passing the `TestSuiteId` object as an argument, to start the suite.
+
+After all the children items within the suite have completed, call `ddtrace.ext.test_visibility.api.TestSuite.finish()`, passing the `TestSuiteId` object as an argument.
```
Comment (urseberry) [inline content/en/tests/setup/python.md:232]:
```suggestion
After all the child items within the suite have completed, call `ddtrace.ext.test_visibility.api.TestSuite.finish()`, passing the `TestSuiteId` object as an argument.
```

File: content/en/tests/setup/python.md [hunk 4]
```diff
@@ -163,6 +163,213 @@ assert True
...
+
+##### Test
+
+A test represents a single test case that is executed as part of a test suite.
+
+Call `ddtrace.ext.test_visibility.api.TestId()`, providing the parent suite's `TestSuiteId`, and the test's name, as arguments to create a `TestId`. The `TestId()` method accepts a JSON-parseable string as the optional `parameters` argument. The `parameters` argument can be used to distinguish parametrized tests that have the same name, but different parameter values.
```
Comment (urseberry) [inline content/en/tests/setup/python.md:238]:
```suggestion
Call `ddtrace.ext.test_visibility.api.TestId()`, providing the parent suite's `TestSuiteId` and the test's name as arguments, to create a `TestId`. The `TestId()` method accepts a JSON-parseable string as the optional `parameters` argument. The `parameters` argument can be used to distinguish parametrized tests that have the same name, but different parameter values.
```

File: content/en/tests/setup/python.md [hunk 5]
```diff
@@ -163,6 +163,213 @@ assert True
...
+
+Call `ddtrace.ext.test_visibility.api.Test.discover()`, passing the `TestId` object as an argument, to discover the test. The `Test.discover()` classmethod accepts a string as the optional `resource` parameter, which default to the `TestId`'s `name`.
```
Comment (urseberry) [inline content/en/tests/setup/python.md:240]:
```suggestion
Call `ddtrace.ext.test_visibility.api.Test.discover()`, passing the `TestId` object as an argument, to discover the test. The `Test.discover()` classmethod accepts a string as the optional `resource` parameter, which defaults to the `TestId`'s `name`.
```

File: content/en/tests/setup/python.md [hunk 6]
```diff
@@ -163,6 +163,213 @@ assert True
...
+- `exc_value`: the `BaseException` object for the exception
+- `exc_traceback`: the `Traceback` object for the exception
+
+###### Codeowner information
+
+The `ddtrace.ext.test_visibility.api.Test.discover()` classmethod accepts an optional list of strings as the `codeowners` parameter.
+
+###### Test source file information
+
+The `ddtrace.ext.test_visibility.api.Test.discover()` classmethod accepts an optional `TestSourceFileInfo` object as the `source_file_info` parameter. A `TestSourceFileInfo` object represents the path and, optionally, the start and end lines for a given test.
```
Comment (urseberry) [inline content/en/tests/setup/python.md:263]:
```suggestion
The `ddtrace.ext.test_visibility.api.Test.discover()` classmethod accepts an optional `TestSourceFileInfo` object as the `source_file_info` parameter. A `TestSourceFileInfo` object represents the path and optionally, the start and end lines for a given test.
```

File: content/en/tests/setup/python.md [hunk 7]
```diff
@@ -163,6 +163,213 @@ assert True
...
+
+#### Code example
+
+```python
+from ddtrace.ext.test_visibility import api
+import pathlib
+import sys
+
+if __name__ == "__main__":
+    # Enabling the Test Visibility service
```
Comment (urseberry) [inline content/en/tests/setup/python.md:286]:
```suggestion
    # Enable the Test Visibility service
```

**Review by urseberry — CHANGES_REQUESTED** [review]
Thank you very much!

File: content/en/tests/setup/python.md (hunk 6 above)
Comment (romainkomorndatadog) [reply content/en/tests/setup/python.md:263]: @urseberry , just double-checking this one. It kinda looks weird to me that there's just the one comma after "optionally".

**Review by urseberry — APPROVED** [review]

**Pull request merged**